api/cluster/EnvironmentRestHandler.go

```diff
@@ -288,7 +288,18 @@ func (impl EnvironmentRestHandlerImpl) GetEnvironmentListForAutocomplete(w http.
 
 		result := impl.enforcer.EnforceByEmailInBatch(emailId, casbin.ResourceGlobalEnvironment, casbin.ActionGet, envIdentifierList)
 		for _, item := range environments {
-			if hasAccess := result[strings.ToLower(item.EnvironmentIdentifier)]; hasAccess {
+
+			var hasAccess bool
+
+			if item.EnvironmentIdentifier != item.ClusterName+"__"+item.Namespace {
+				// fix for futuristic case
+				rbacObject2 := item.ClusterName + "__" + item.Namespace
+				hasAccess = result[strings.ToLower(rbacObject2)] || result[strings.ToLower(item.EnvironmentIdentifier)]
```

Comment:
if rbacObject2 is to needed everywhere in lowerCase why not do - rbacObject2 := strings.ToLower(item.ClusterName + "__" + item.Namespace)

Reply:
same for environmentIdentifier. can be done like environmentIdentifier := strings.ToLower(item.EnvironmentIdentifier) or item.EnvironmentIdentifier = strings.ToLower(item.EnvironmentIdentifier).